From b244f277c00e57277ccd3aa2135da2c9bb44af8f Mon Sep 17 00:00:00 2001
From: Kihwal Lee <kihwal@apache.org>
Date: Fri, 9 Dec 2016 11:05:04 -0600
Subject: [PATCH 2224/2748] HADOOP-11859. PseudoAuthenticationHandler fails
 with httpcomponents v4.4. Contributed by Eugene
 Koifman.

(cherry picked from commit db4419df6bc72e9a1ddd90bac4dd7dcf0fb05479)
(cherry picked from commit deae1f83239c3f1d3ac44bda6a6fd90b509f5dfa)

 Conflicts:
	hadoop-common-project/hadoop-common/CHANGES.txt

Change-Id: I4c5ed80eb776f3e72d16b40a684ac7dabc0d639b
---
 .../server/PseudoAuthenticationHandler.java        |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/hadoop-common-project/hadoop-auth/src/main/java/org/apache/hadoop/security/authentication/server/PseudoAuthenticationHandler.java b/hadoop-common-project/hadoop-auth/src/main/java/org/apache/hadoop/security/authentication/server/PseudoAuthenticationHandler.java
index 50f0cf1..19947aa 100644
--- a/hadoop-common-project/hadoop-auth/src/main/java/org/apache/hadoop/security/authentication/server/PseudoAuthenticationHandler.java
+++ b/hadoop-common-project/hadoop-auth/src/main/java/org/apache/hadoop/security/authentication/server/PseudoAuthenticationHandler.java
@@ -142,7 +142,11 @@ public boolean managementOperation(AuthenticationToken token,
   }
 
   private String getUserName(HttpServletRequest request) {
-    List<NameValuePair> list = URLEncodedUtils.parse(request.getQueryString(), UTF8_CHARSET);
+    String queryString = request.getQueryString();
+    if(queryString == null || queryString.length() == 0) {
+      return null;
+    }
+    List<NameValuePair> list = URLEncodedUtils.parse(queryString, UTF8_CHARSET);
     if (list != null) {
       for (NameValuePair nv : list) {
         if (PseudoAuthenticator.USER_NAME.equals(nv.getName())) {
-- 
1.7.9.5

